File: Volume_000/0016_Treasure_Hunt.rb
x, y = 0, 0
a = 90

while line = gets
  break if line.chomp == '0,0'
  c, d = line.chomp.split(',').map(&:to_i)
  x += c * Math.cos(a * Math::PI / 180)
  y += c * Math.sin(a * Math::PI / 180)
  a -= d
end

puts x.to_i, y.to_i

File: Volume_000/0016_Treasure_Hunt.cpp
#include <iostream>
#include <math.h>
#include <cstdio>

using namespace std;

int main() {
    double x, y, current_angle;
    int step, angle;
    x = 0, y = 0, current_angle = 90;

    while (true) {
        scanf("%d,%d", &step, &angle);

        if (!step && !angle) {
            break;
        }

        x += step * cos(current_angle * M_PI / 180);
        y += step * sin(current_angle * M_PI / 180);
        current_angle -= angle;
    }

    printf("%d\n%d\n", (int)x, (int)y);

    return 0;
}
